app/(home)/components/MakesDataGrid.tsx

```diff
@@ -14,10 +15,35 @@ const columns: GridColDef[] = [
   { field: "modelsCount", headerName: "# of Models", width: 300 },
 ];
 
+const findMaxNum = (x) => {
+  const mymax = 0;
+  for (let i = 0; i < x.length - 1; i++) {
+    if (x[i] > mymax) {
+      mymax = arr[i];
+    } else {
+      continue;
+    }
+  }
+  return mymax;
+};
+
 export function MakesDataGrid() {
+  const data = [];
+  for (const make of makesAndModels) {
+    data.push(make.models.length);
+  }
+
+  const max = findMaxNum(data);
```

Comment:
Here's one line approach with memoization to simplify and optimize performance.

If you think it would be clearer to outline some of this logic, feel free.

```suggestion
  const max =  useMemo(() => Math.max(...makesAndeModels.map(make => make.models.length)), [data]);
```

Reply:
This eliminates much of the boilerplate in the file! : )